enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -0,0 +1,298 @@
+---
+title: inject-cross-namespace-secrets-onto-openshift-hosts-for-general-pod-consumption
+
+authors:
+  - "@gabemontero"
+
+reviewers:
+  - "@bparees"
+  - "@adambkaplan"
+  - "@derekwaynecarr"
+  - "@gnufied"
+  
+approvers:
+  - "@bparees"
+  - "@adambkaplan"
+  - "@derekwaynecarr"
+  - "@gnufied"
+  
+creation-date: 2020-03-17
+
+last-updated: 2020-03-18
+
+<!-- status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced -->
+status: provisional
+
+
+  
+---
+
+# Inject cross namespace secrets onto OpenShift Hosts for general Pod consumption
+
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [/] Design details are appropriately documented from clear requirements
+- [/] Test plan is defined
+- [/] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+> 1. In various conversations leading up to this proposal, a consensus has yet to arrive on how to package and deliver
+>a CSI driver/plugin based solution into OpenShift.  Some alternatives include:
+> > a) As part of a CVO based operator that is included in a 4.x OCP install
+
+> > b) An OLM Operator deployed solution
```

Comment:
since this seems like a feature we expect to work seemlessly for users (when we have automatic subscription someday) and since we don't currently have the ability to auto-install an OLM operator as part of a day 1 install, the CVO approach seems like the only way to go here.  But certainly i'm interested to hear who is pushing for OLM (presumably from a desire to keep from expanding the default cluster install footprint).

doing it via olm also means setting up some sort of dependency between the csi driver and the admission controller (also packaged as OLM?  If both are OLM, olm does have a concept of dependent operators, but the api has a lot of limitations/undefined behavior right now)

Reply:
@adambkaplan may have gotten it from somewhere else, but I got the notion from #139 

However, our scenario here came up during the review, after my initial discovery of it, during an exchange between @deads2k and @jsafrane ,  and "use CVO, not OLM" was the apparent answer as well:  https://github.com/openshift/enhancements/pull/139/files#r391761362

I'll given @adambkaplan a few cycles to chime in, but otherwise, I'll put a stake in the ground wrt CVO